dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java

```diff
@@ -269,6 +277,77 @@ public void findAllTest() throws Exception {
         ;
     }
 
+    @Test
+    public void findAllNoDuplicatesOnMultipleCommunityTitlesTest() throws Exception {
```

Comment:
Could we also add in a pagination-related integration test?  As @abollini pointed out, the prior implementation in this PR would have failed to provide proper pagination if (for example) we had page size set to 5, and had 6 total community dc.titles, but where two dc.titles were on the same Community.  In that situation, we would have seen 2 pages, but (once deduplicated) only 5 Communities (which should mean it's 1 page).

So, I think we should prove out here that pagination is correct -- it should be.  But, a simple test would be to set page size to 2, create 2 communities (where at least one has multiple titles), and ensure that we end up with 2 total results on 1 page.